### PR TITLE
[Keyboard] Add a new keyboard

### PR DESCRIPTION
**Pull request by ningjx**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ning/tiny_board/tb16_rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 4
+#define MATRIX_COLS 4
+
+#define TAPPING_TERM 200
+
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/config.h:24]:
```suggestion
```

File: keyboards/ning/tiny_board/tb16_rgb/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# ning/tiny_board/tb16_rgb
+
+![ning/tiny_board/tb16_rgb](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [Ning](https://github.com/ningjx)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/readme.md:9]: Please fill this out.

File: keyboards/ning/tiny_board/tb16_rgb/keymaps/ningjx/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    [0] = LAYOUT_ortho_4x4(
+        KC_P7,   KC_P8,   KC_P9,   RGB_TOG,
+        KC_P4,   KC_P5,   KC_P6,   RGB_MODE_FORWARD,
+        KC_P1,   KC_P2,   KC_P3,   KC_PMNS,
+        KC_P0,   KC_PDOT, KC_PENT, KC_PPLS
+    )
+};
+
+led_config_t g_led_config = { {
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/keymaps/ningjx/keymap.c:39]: This does not belong here.

File: keyboards/ning/tiny_board/tb16_rgb/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    [0] = LAYOUT_ortho_4x4(
+        KC_P7,   KC_P8,   KC_P9,   RGB_TOG,
+        KC_P4,   KC_P5,   KC_P6,   RGB_MODE_FORWARD,
+        KC_P1,   KC_P2,   KC_P3,   KC_PMNS,
+        KC_P0,   KC_PDOT, KC_PENT, KC_PPLS
+    )
+};
+
+led_config_t g_led_config = { {
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/keymaps/default/keymap.c:39]: As above.

File: keyboards/ning/tiny_board/tb16_rgb/rules.mk [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# This file intentionally left blank
+MCU = atmega32u4
+BOOTLOADER = caterina
+RGB_MATRIX_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/rules.mk:4]:
```suggestion
```

File: keyboards/ning/tiny_board/tb16_rgb/info.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    "maintainer": "ningjx",
+    "bootloader": "caterina",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (fauxpark) [inline keyboards/ning/tiny_board/tb16_rgb/info.json:13]:
```suggestion
        "nkro": true,
        "rgb_matrix": true
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ning/tiny_board/tb16_rgb/info.json (hunk 6 above)
Comment (ningjx) [reply keyboards/ning/tiny_board/tb16_rgb/info.json]: I have fixed all your suggestions,can you help me to review my code again,thanks

File: keyboards/ning/tiny_board/tb16_rgb/rules.mk [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = bitbang
```
Comment (waffle87) [inline keyboards/ning/tiny_board/tb16_rgb/rules.mk:3]:
```suggestion
RGB_MATRIX_DRIVER = WS2812
```

File: keyboards/ning/tiny_board/tb16_rgb/readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# ning/tiny_board/tb16_rgb
+
+![ning/tiny_board/tb16_rgb](imgur.com image replace me!)
```
Comment (waffle87) [inline keyboards/ning/tiny_board/tb16_rgb/readme.md:3]: This should be filled out or removed.

**Review by waffle87 — APPROVED** [review]

File: keyboards/ning/tiny_board/tb16_rgb/info.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "matrix_pins": {
+        "cols": ["B6", "B1", "B3", "B2"],
+        "rows": ["F7", "F6", "F5", "F4"]
+    },
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/ning/tiny_board/tb16_rgb/info.json:20]:
```suggestion
    "development_board": "promicro",
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**